alacritty/src/config/ui_config.rs

```diff
@@ -473,11 +474,102 @@ pub struct HintBinding {
 /// Hint mouse highlighting.
 #[derive(ConfigDeserialize, Default, Copy, Clone, Debug, PartialEq, Eq)]
 pub struct HintMouse {
-    /// Hint mouse highlighting availability.
-    pub enabled: bool,
-
     /// Required mouse modifiers for hint highlighting.
     pub mods: ModsWrapper,
+
+    /// Mouse button which triggers the command. None to disable mouse hints
+    #[config(alias = "enabled")]
+    pub button: HintMouseButton,
```

Comment:
Have you tried `Option<HintMouseButton>` instead of making it a dedicated variant? The derive macro should automatically support that iirc. That way you can maybe save yourself some work in deserializing none explicitly.